### PR TITLE
Orange: add volume definition and JSON I/O for surface/volume

### PR DESCRIPTION
**Pull request by sethrj**
Add volume definitions, sample geometries, and JSON-based I/O implementations for surface and volume reading in ORANGE.

### CODE REVIEW

**Review by amandalund — APPROVED** [review]
Looks great @sethrj! Nothing from me :)

File: src/orange/Data.hh [hunk 1]
```diff
@@ -59,5 +62,145 @@ struct SurfaceData
...
+{
+    //// TYPES ////
+
+    template<class T>
+    using Items = Collection<T, W, M, UniverseId>;
+
+    //// DATA ////
+
+    //// METHODS ////
+
```
[the remark this answers is not in the document]
Comment (sethrj) [reply src/orange/Data.hh:143]: Yep indeed. I thought I left it out of this one 🤔 so I'll double check that it belongs here. Thanks!

[the remark this answers is not in the document]
Comment (sethrj) [reply src/orange/Data.hh:143]: Yeah ok, it's just a placeholder. Since it might not be used for a fair bit, I could just delete it.

**Comment by sethrj** [comment]
CI failure is another issue with disk space...

**Pull request merged**